Donation-platform/src/main/java/com/google/FindTrendingCharities.java

```diff
@@ -46,31 +47,39 @@
     final double charityTagsWeight = 0.75;
     final double avgReviewWeight = 0.25;
 
-
+    
     //returns the collection of top trending charities
     public Collection<Charity> query() {
+        DbSetUpUtils setUp = new DbSetUpUtils();
+        db = setUp.getDbCalls();
+        //setUp.dbSetUp();                                      //only call once
         //TODO: Change to getAllCharities() when integrating db
-        Collection<Charity> charities = getAllHardCodedCharities();
+        Collection<Charity> charities = getAllCharities();
         for (Charity cur: charities) {
             double curScore = calcCharityTrendingScore(cur);
-            cur.setTrendingScore(curScore);
+            cur.setTrendingScoreCharity(curScore);
         }
         ArrayList<Charity> charitiesList = new ArrayList<>(charities);
-        Collections.sort(charitiesList, new SortByTrending());
+        Collections.sort(charitiesList);
         List<Charity> topTrending = charitiesList.subList(0, trendingNum);
```

Comment:
check if length of charitiesList is smaller than trendingNum

Reply:
done